src/posix/main.c

```diff
@@ -358,6 +358,11 @@ static const otCliCommand kCommands[] = {
     {"netif", ProcessNetif},
 };
 
+#if OPENTHREAD_CONFIG_CLI_VENDOR_COMMANDS_ENABLE
```

Comment:
Add `OPENTHREAD_CONFIG_CLI_VENDOR_COMMANDS_ENABLE` in `cli/cli_config.h` and set to zero if not defined?
```
#ifndef OPENTHREAD_CONFIG_CLI_VENDOR_COMMANDS_ENABLE
#define OPENTHREAD_CONFIG_CLI_VENDOR_COMMANDS_ENABLE 0
#endif
```

Some toolchains check/require that all macors are defined.